src/metabase/driver/sql_jdbc/sync/describe_database.clj

```diff
@@ -206,15 +215,11 @@
       (let [schema-filter-prop      (driver.u/find-schema-filters-prop driver)
             has-schema-filter-prop? (some? schema-filter-prop)
             database                (db-or-id-or-spec->database db-or-id-or-spec)
-            default-active-tbl-fn   #(into #{} (sql-jdbc.sync.interface/active-tables driver conn nil nil))]
-        (if has-schema-filter-prop?
-          ;; TODO: the else of this branch seems uncessary, why do you want to call describe-database on a database that
-          ;; does not exists?
-          (if (some? database)
-            (let [prop-nm                                 (:name schema-filter-prop)
-                  [inclusion-patterns exclusion-patterns] (driver.s/db-details->schema-filter-patterns
-                                                           prop-nm
-                                                           database)]
-              (into #{} (sql-jdbc.sync.interface/active-tables driver conn inclusion-patterns exclusion-patterns)))
-            (default-active-tbl-fn))
-          (default-active-tbl-fn)))))})
+            [inclusion-patterns
+             exclusion-patterns]    (when (and has-schema-filter-prop? (some? database))
+                                      (driver.s/db-details->schema-filter-patterns
+                                       (:name schema-filter-prop)
+                                       database))]
+        (->> (sql-jdbc.sync.interface/active-tables driver conn inclusion-patterns exclusion-patterns)
+             (tables-with-properties driver database)
```

Comment:
I'm not fond of `join` as it makes me think of join in SQL, wdyt about `hydrate-table-properties`? `add` would be okay to IMO.

Reply:
I'm not super keen on `hydrate`, as we we use that term for toucan models. How about `calculate-table-properties`? Still fine with `add` as well, but thinking now it's nice to indicate that this function "does some work".

Reply:
I'm not sure about `calculate` since to me it means doing some arithmetic calculation, but here we just fetch the row count from DB.

That's said I went with `add-table-properties` here :)

Reply:
Fair enough. For me touching a database is doing work 😄